/**
 * This code was GENERATED using the solita package.
 * Please DO NOT EDIT THIS FILE, instead rerun solita to update it or write a wrapper to add functionality.
 *
 * See: https://github.com/metaplex-foundation/solita
 */

type ErrorWithCode = Error & { code: number }
type MaybeErrorWithCode = ErrorWithCode | null | undefined

const createErrorFromCodeLookup: Map<number, () => ErrorWithCode> = new Map()
const createErrorFromNameLookup: Map<string, () => ErrorWithCode> = new Map()

/**
 * DigitalIdentityDoesntMatchWithAuth: 'Digital Identity Account cannot be verified'
 *
 * @category Errors
 * @category generated
 */
export class DigitalIdentityDoesntMatchWithAuthError extends Error {
  readonly code: number = 0x1770
  readonly name: string = 'DigitalIdentityDoesntMatchWithAuth'
  constructor() {
    super('Digital Identity Account cannot be verified')
    if (typeof Error.captureStackTrace === 'function') {
      Error.captureStackTrace(this, DigitalIdentityDoesntMatchWithAuthError)
    }
  }
}

createErrorFromCodeLookup.set(
  0x1770,
  () => new DigitalIdentityDoesntMatchWithAuthError()
)
createErrorFromNameLookup.set(
  'DigitalIdentityDoesntMatchWithAuth',
  () => new DigitalIdentityDoesntMatchWithAuthError()
)

/**
 * DigitalProofDoesntMatchWithAuth: 'Digital Proofs Account cannot be verified'
 *
 * @category Errors
 * @category generated
 */
export class DigitalProofDoesntMatchWithAuthError extends Error {
  readonly code: number = 0x1771
  readonly name: string = 'DigitalProofDoesntMatchWithAuth'
  constructor() {
    super('Digital Proofs Account cannot be verified')
    if (typeof Error.captureStackTrace === 'function') {
      Error.captureStackTrace(this, DigitalProofDoesntMatchWithAuthError)
    }
  }
}

createErrorFromCodeLookup.set(
  0x1771,
  () => new DigitalProofDoesntMatchWithAuthError()
)
createErrorFromNameLookup.set(
  'DigitalProofDoesntMatchWithAuth',
  () => new DigitalProofDoesntMatchWithAuthError()
)

/**
 * Attempts to resolve a custom program error from the provided error code.
 * @category Errors
 * @category generated
 */
export function errorFromCode(code: number): MaybeErrorWithCode {
  const createError = createErrorFromCodeLookup.get(code)
  return createError != null ? createError() : null
}

/**
 * Attempts to resolve a custom program error from the provided error name, i.e. 'Unauthorized'.
 * @category Errors
 * @category generated
 */
export function errorFromName(name: string): MaybeErrorWithCode {
  const createError = createErrorFromNameLookup.get(name)
  return createError != null ? createError() : null
}
